packages/itwin/ec3-widget/src/components/ReportTableSelector.tsx

```diff
@@ -0,0 +1,119 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+
+import { IModelApp } from "@itwin/core-frontend";
+import { ComboBox, Label, toaster } from "@itwin/itwinui-react";
+import React, { useCallback, useEffect, useMemo, useState } from "react";
+import useValidator, { NAME_REQUIREMENTS } from "../hooks/useValidator";
+import type { Configuration } from "./Template";
+import "./ReportTableSelector.scss";
+import { useODataClient } from "./api/context/ODataClientContext";
+
+export interface ReportTableSelectorProps {
+  selectedReportTable: string;
+  template: Configuration;
+  placeHolder: string;
+  onChange: (table: string, numCols: string[], strCols: string[]) => Promise<void>;
+  setLoading: (val: boolean) => void;
+}
+
+export const ReportTableSelector = ({
+  selectedReportTable,
+  template,
+  placeHolder,
+  onChange,
+  setLoading,
+}: ReportTableSelectorProps) => {
+  const [validator, _showValidationMessage] = useValidator();
+  const [reportTable, setReportTable] = useState(selectedReportTable);
+  const [reportTables, setReportTables] = useState<string[] | undefined>(undefined);
+  const reportingClientApi = useODataClient();
+
+  const updateData = useCallback(async (reportTableName: string) => {
+    if (!IModelApp.authorizationClient)
+      throw new Error(
+        "AuthorizationClient is not defined. Most likely IModelApp.startup was not called yet."
+      );
+    if (!template.reportId)
```

Comment:
What is the type of `reportId`? I don't think it can be falsy. Unless it's an empty string. Can it be an empty string?

Reply:
`reportId` should never be an empty string. It should either be `null` or `undefined`.

Reply:
made reportId nullable and instead of empty string it is now undefined